src/ParameterDistributions.jl

```diff
@@ -160,39 +161,76 @@ n_samples(d::Parameterized) = "Distribution stored in Parameterized form, draw s
 Structure to hold a parameter distribution, always stored as an array of distributions.
 """
 struct ParameterDistribution{PDType <: ParameterDistributionType, CType <: ConstraintType, ST <: AbstractString}
-    distributions::Array{PDType}
-    constraints::Array{CType}
-    names::Array{ST}
-
-    function ParameterDistribution(
-        parameter_distributions::Union{PDType, Array{PDType}},
-        constraints::Union{CType, Array{CType}, Array},
-        names::Union{ST, Array{ST}},
-    ) where {PDType <: ParameterDistributionType, CType <: ConstraintType, ST <: AbstractString}
-
-        parameter_distributions =
-            isa(parameter_distributions, PDType) ? [parameter_distributions] : parameter_distributions
-        n_parameters_per_dist = [ndims(pd) for pd in parameter_distributions]
-        constraints = isa(constraints, Union{<:ConstraintType, Array{<:ConstraintType}}) ? [constraints] : constraints #to calc n_constraints_per_dist
-        names = isa(names, ST) ? [names] : names
-
-        n_constraints_per_dist = [length(c) for c in constraints]
-        n_dists = length(parameter_distributions)
-        n_names = length(names)
-        if !(n_parameters_per_dist == n_constraints_per_dist)
-            throw(DimensionMismatch("There must be one constraint per parameter in a distribution, use no_constraint() type if no constraint is required"))
-        elseif !(n_dists == n_names)
-            throw(DimensionMismatch("There must be one name per parameter distribution"))
-        else
-            constraints = cat(constraints..., dims = 1)
+    distributions::AbstractVector{PDType}
+    constraints::AbstractVector{CType}
+    names::AbstractVector{ST}
+end
+
+"""
+    ParameterDistribution(param_dist_dict::Union{Dict,AbstractVector})
+
+Constructor taking in a Dict or array of Dicts. Each dict must contain the key-val pairs:
+- `"distribution"` - a distribution of `ParameterDistributionType`
+- `"constraints"` - constraint(s) given as a `ConstraintType` or array of `ConstraintType`s with length equal to the dims of the distribution
+- `"name"` - a name of the distribution as a String.
+"""
+function ParameterDistribution(param_dist_dict::Union{Dict, AbstractVector})
 
-            new{PDType, ConstraintType, ST}(parameter_distributions, constraints, names)
+    #check type
+    if !isa(param_dist_dict, Dict)
+        if !(eltype(param_dist_dict) <: Dict)
+            throw(ArgumentError("input argument must be a Dict, or <:AbstractVector{Dict}"))
         end
     end
 
-end
+    #make copy as array
+    param_dist_dict_array = !isa(param_dist_dict, AbstractVector) ? [param_dist_dict] : param_dist_dict
+    #perform checks on the individual distributions
+    for pdd in param_dist_dict_array
+        #check all keys are present
+        if !all(k in collect(keys(pdd)) for k in ["distribution", "name", "constraints"])
```

Comment:
```suggestion
        if !all(keys(pdd) .∈ [["distribution", "name", "constraints"]])
```

Reply:
I think this should be the other way around, we check if "[distribution]... form a subset of keys(pdd)" rather than the other way around. But otherwise i can compact it.

Reply:
done